buildpack/files.go

```diff
@@ -197,52 +193,14 @@ func containsName(unmet []Unmet, name string) bool {
 
 // layer content metadata
 
-type EncoderDecoder interface {
-	IsSupported(buildpackAPI string) bool
-	Encode(file *os.File, lmf layertypes.LayerMetadataFile) error
-	Decode(path string) (layertypes.LayerMetadataFile, string, error)
+type LayersMetadata struct {
+	ID      string                   `json:"key" toml:"key"`
+	Version string                   `json:"version" toml:"version"`
+	Layers  map[string]LayerMetadata `json:"layers" toml:"layers"`
+	Store   *StoreTOML               `json:"store,omitempty" toml:"store"`
 }
 
-func defaultEncodersDecoders() []EncoderDecoder {
-	return []EncoderDecoder{
-		// TODO: it's weird that api05 is relevant for buildpack APIs 0.2-0.5 and api06 is relevant for buildpack API 0.6 and up. We should work on it.
-		api05.NewEncoderDecoder(),
-		api06.NewEncoderDecoder(),
-	}
-}
-
-func EncodeLayerMetadataFile(lmf layertypes.LayerMetadataFile, path, buildpackAPI string) error {
-	fh, err := os.Create(path)
-	if err != nil {
-		return err
-	}
-	defer fh.Close()
-
-	encoders := defaultEncodersDecoders()
-
-	for _, encoder := range encoders {
-		if encoder.IsSupported(buildpackAPI) {
-			return encoder.Encode(fh, lmf)
-		}
-	}
-	return errors.New("couldn't find an encoder")
-}
-
-func DecodeLayerMetadataFile(path, buildpackAPI string) (layertypes.LayerMetadataFile, string, error) { // TODO: pass the logger and print the warning inside (instead of returning a message)
-	fh, err := os.Open(path)
-	if os.IsNotExist(err) {
-		return layertypes.LayerMetadataFile{}, "", nil
-	} else if err != nil {
-		return layertypes.LayerMetadataFile{}, "", err
-	}
-	defer fh.Close()
-
-	decoders := defaultEncodersDecoders()
-
-	for _, decoder := range decoders {
-		if decoder.IsSupported(buildpackAPI) {
-			return decoder.Decode(path)
-		}
-	}
-	return layertypes.LayerMetadataFile{}, "", errors.New("couldn't find a decoder")
+type LayerMetadata struct {
+	SHA string `json:"sha" toml:"sha"`
+	layermetadata.File
```

Comment:
Now that we have the `buildpack/layermetadata/file.go`, all the structs seem like they belong together. It's just a bit awkward this logic spread out.

Reply:
@aemengo your comment made me wonder why we need the `layermetadata` package to begin with. WDYT of 63a91a52 ?

Reply:
I like it. 👍🏾

Reply:
Fixed in 1656b02e